native_client/deepspeech.cc

```diff
@@ -428,8 +477,37 @@ ModelState::decode(vector<float>& logits)
   vector<Output> out = ctc_beam_search_decoder(
     inputs.data(), n_frames, num_classes, *alphabet, beam_width,
     cutoff_prob, cutoff_top_n, scorer);
+	
+  return out;
+}
 
-  return strdup(alphabet->LabelsToString(out[0].tokens).c_str());
+Metadata* ModelState::decode_metadata(vector<float>& logits) 
+{
+  vector<Output> out = decode_raw(logits);
+	
+  Metadata* metadata = (Metadata*)malloc(sizeof (Metadata));
+  metadata->num_items = out[0].tokens.size();
+  metadata->items = (MetadataItem*)malloc(sizeof(MetadataItem) * metadata->num_items);
+
+  // Loop through each character
+  for (int i = 0; i < out[0].tokens.size(); ++i) {
+    char* character = (char*)alphabet->StringFromLabel(out[0].tokens[i]).c_str();
+
+    // Note: 1 timestep = 20ms
+	// Timesteps are from the position of the highest letter probability so we 
+	// offset them back by 3ms to account for this
+    float start_time = static_cast<float>(out[0].timesteps[i] * 0.02 - 0.03);
+
+    metadata->items[i] = (MetadataItem) { 
+      .character = character, 
+      .timestep = out[0].timesteps[i], 
+      .start_time = start_time
+    };
```

Comment:
Designated initializers aren't available in C++11. I think GCC and Clang may support it as extensions but it'll probably break our Windows build. Better to go with the old ugly way of initializing each member individually.